### ISSUE TITLE
Stage 1 decorators issue

### ISSUE THREAD

**Issue by qmmr** [babel/babel#1350]
The example from [release notes](http://babeljs.io/blog/2015/03/31/5.0.0/) is not working for me with `babel --version // 5.1.13`

.babelrc
```json
{
	"stage": 0,
	"optional": [
		"es7.decorators"
	]
}
```
decorators.js
```js
function concat(...args) {
  let sep = args.pop();

  return function(target, key, descriptor) {
    descriptor.initializer = function() {
      return args.map(arg => this[arg]).join(sep);
    }
  }
}

function autobind(target, key, descriptor) {
  var fn = descriptor.value;
  delete descriptor.value;
  delete descriptor.writable;
  descriptor.get = function () {
    return this[key] = fn.bind(this);
  };
}

class Person {
  firstName = "Sebastian";
  lastName = "McKenzie";

  @concat("firstName", "lastName", " ") fullName;
  @concat("lastName", "firstName", ", ") formalName;

  @autobind
  getFullName() {
    return `${this.first} ${this.last}`;
  }
}

console.log(new Person().firstName)
console.log(new Person().lastName)
console.log(new Person().fullName)
console.log(new Person().formalName)
console.log(new Person().getFullName)

assert(new Person().fullName, "Sebastian McKenzie");
assert(new Person().formalName, "McKenzie, Sebastian");
assert(new Person().getFullName.call(null), "Sebastian McKenzie");
```

When after compilation `babel descriptors.js -o compiled.js` and running `node compiled.js` I'm getting the following error
```bash
$ node scripts/compileds.js
Sebastian
McKenzie

,
/Users/marcin.kumorek/projects/es6-today/scripts/compileds.js:32
    return this[key] = fn.bind(this);
                     ^
TypeError: Cannot set property getFullName of #<Person> which has only a getter
    at Person.descriptor.get (/Users/marcin.kumorek/projects/es6-today/scripts/compileds.js:32:22)
    at Object.<anonymous> (/Users/marcin.kumorek/projects/es6-today/scripts/compileds.js:75:25)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Function.Module.runMain (module.js:501:10)
    at startup (node.js:129:16)
    at node.js:814:3
```
There are two issues here:
- first the output from console.logs are wrong (fullName returns '' and formalName ',')
- second the @autobind is not working correctly resulting in a TypeError

I'm running `node -v0.12`

**Comment by sebmck** [comment]
See #1310.

**Comment by qmmr** [comment]
@sebmck Two questions:
1. If I remove the @autobind, I'm still having issue with fullName and formalName
2. It is confusing for people to see the example that's not working. Are planning to change it to something more usable?

**Comment by sebmck** [comment]
@qmmr 

> If I remove the @autobind, I'm still having issue with fullName and formalName

What issues?

> It is confusing for people to see the example that's not working. Are you planning to change it to something more usable?

I'll happily take a PR for [babel.github.io/_posts/2015-03-31-5.0.0.md](https://github.com/babel/babel.github.io/blob/master/_posts/2015-03-31-5.0.0.md). I am stretched extremely thin and cannot do everything.

**Comment by qmmr** [comment]
@sebmck Oh, sorry, I did write it all together.

I run `babel descriptors.js -o compiled.js` and then `node compiled.js` The output:

```bash
$ node scripts/compileds.js
Sebastian
McKenzie

,
```
But it should have been.
```bash
$ node scripts/compileds.js
Sebastian
McKenzie
Sebastian McKenzie
McKenzie, Sebastian
```
I have replaced the `assert` with `console.logs` from the example.

**Comment by monsanto** [comment]
@sebmck @qmmr I believe this is a bug in Babel. [The property initializers spec](https://github.com/wycats/javascript-decorators/blob/master/INITIALIZER_INTEROP.md#property-initializers) suggests that `descriptor.initalizer` should be invoked with the target as `this`; currently `this` will be `descriptor`.

**Comment by sebmck** [comment]
Sorry for the delay, fixed as of 5.2.0. Thanks!
